js_modules/dagster-ui/packages/ui-core/src/assets/useAllAssets.tsx

```diff
@@ -0,0 +1,278 @@
+    [materializedAssets, allAssetNodesByKey, allAssetNodes, groupSelector],
+  );
+
+  const assetsByAssetKey = useMemo(() => getAssetsByAssetKey(assets), [assets]);
+
+  return {
+    assets,
+    loading: loading || workspaceLoading,
+    query: manager.fetchAssets,
+    assetsByAssetKey,
+    error,
+  };
+}
+
+const getFetchManager = weakMapMemoize((client: ApolloClient<any>) => new FetchManager(client));
+
+const VERSION = 1;
+class FetchManager {
+  private _assetsOrError: AssetRecord[] | PythonErrorFragment | null = null;
+  private _subscribers = new Set<(assetsOrError: AssetRecord[] | PythonErrorFragment) => void>();
+  private _started = false;
+  private _fetchTimeout: NodeJS.Timeout | null = null;
+  private _fetchPromise: Promise<AssetRecord[] | PythonErrorFragment> | null = null;
+  private _batchLimit = DEFAULT_BATCH_LIMIT;
+  private _cache: ReturnType<typeof cache<CacheData<AssetRecord[]>>>;
+
+  constructor(private readonly client: ApolloClient<any>) {
+    this._cache = cache<CacheData<AssetRecord[]>>({dbName: 'MaterializedAssets', maxCount: 1});
+    this.loadFromIndexedDB(); // Load from IndexedDB on construction, intentionally not awaited.
+  }
+
+  subscribe(callback: (assetsOrError: AssetRecord[] | PythonErrorFragment) => void) {
+    this._subscribers.add(callback);
+    this.startFetchLoop();
+
+    if (this._assetsOrError) {
+      callback(this._assetsOrError);
+    }
+
+    return () => {
+      this._subscribers.delete(callback);
+      if (!this._subscribers.size) {
+        this.stopFetchLoop();
+      }
+    };
+  }
+
+  private async loadFromIndexedDB() {
+    const result = await this._cache.get('data');
+    if (!result) {
+      return;
+    }
+    const {data, version} = result.value;
+    if (data && !this._assetsOrError && version === VERSION) {
+      this._assetsOrError = data;
+      this._subscribers.forEach((callback) => callback(data));
+    }
+  }
+
+  private saveToIndexedDB(data: AssetRecord[]) {
+    this._cache.set('data', {data, version: VERSION});
+  }
+
+  private startFetchLoop() {
+    if (this._started) {
+      return;
+    }
+    this._started = true;
+    this.fetchAssets();
+  }
+
+  private stopFetchLoop() {
+    this._started = false;
+    if (this._fetchTimeout) {
+      clearTimeout(this._fetchTimeout);
+      this._fetchTimeout = null;
+    }
+  }
+
+  public fetchAssets = async (pollInterval = POLL_INTERVAL) => {
+    if (this._fetchPromise) {
+      return this._fetchPromise;
+    }
+    let nextAssetsOrError: AssetRecord[] | PythonErrorFragment | null = null;
+    try {
+      this._fetchPromise = fetchAssets(this.client, this._batchLimit);
+      nextAssetsOrError = await this._fetchPromise;
+      this._assetsOrError = nextAssetsOrError;
+    } finally {
+      this._fetchPromise = null;
+    }
+
+    let nextPollInterval = POLL_INTERVAL;
+    if (nextAssetsOrError instanceof Array) {
+      setTimeout(() => {
+        this.saveToIndexedDB(nextAssetsOrError);
+      }, 32);
```

Comment:
is there significance to this number? It feels like there is given the specificity

Reply:
Oh it's just 1 30fps frame away